go-controller/pkg/testing/testing.go

```diff
@@ -82,7 +142,14 @@ func (f *FakeExec) AddFakeCmd(expected *ExpectedCmd) {
 				},
 			},
 		}
-	})
+	}
+	expectedCommandline, _ := getExpectedCommandline(expected.Cmd)
+	f.expectedCommands = append(f.expectedCommands, expectedCommandline)
+	if f.looseCompare {
+		f.commandPool[expectedCommandline] = kCmd
```

Comment:
@alexanderConstantinescu does this assume that every instance of the same executed command will return the same result function during loose compare? eg I don't think we'd be able to loose compare something that polls getting port addresses a couple times and then gets the right answer on the 4th try?

Reply:
Well, you're right. Unfortunately `Command` is called using `cmd` and `args`, meaning we can only lookup any expected executed command by these fields. So if we have a situation where we expect the same command to be called twice and return different outputs for each, then this method won't help. We would be forced to use the regular ordered method of comparison instead. I think this is what you meant right?